src/compas_fab/backends/backend_feature_interfaces.py

```diff
@@ -0,0 +1,116 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from abc import ABCMeta
+from abc import abstractmethod
+
+
+class ForwardKinematics(object):
+    __metaclass__ = ABCMeta
+
+    def __call__(self, configuration, group=None, options={}):
```

Comment:
I think the empty dictionary is problematic here. Because it's the "same" everywhere, so, if any of these methods modify it, it will affect everything else, if should default to `None` and init to `{}` inside.